modules/messages/src/lib.rs

```diff
@@ -349,18 +368,38 @@ pub mod pallet {
 					let (unspent_weight, refund_pay_dispatch_fee) = match receival_result {
 						ReceivalResult::Dispatched(dispatch_result) => {
 							valid_messages += 1;
+							lane_messages_received_status
+								.push((message.key.nonce, ReceivedMessageResult::Dispatched));
 							(
 								dispatch_result.unspent_weight,
 								!dispatch_result.dispatch_fee_paid_during_dispatch,
 							)
 						},
-						ReceivalResult::InvalidNonce |
-						ReceivalResult::TooManyUnrewardedRelayers |
-						ReceivalResult::TooManyUnconfirmedMessages => (dispatch_weight, true),
+						ReceivalResult::InvalidNonce => {
```

Comment:
Maybe we can have `impl From<ReceivalResult>` for `NotDispatchedReason` => less code

Reply:
thats what I had at first :D and made `pub struct ReceivedMessages<Result> {`  generic,
but then I added `NotEnoughWeightForLane`  so I better created new enum/struct,
but now when I removed `NotEnoughWeightForLane`  I can use `ReceivalResult` :D

ok, no problem, I will clear that :)